src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,103 @@
-export const Pagination = () => {};
+import { FC, useEffect, useState } from 'react';
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+interface Props {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void,
+}
+
+export const Pagination: FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const [chosenPage, setChosenPage] = useState(currentPage);
+
+  useEffect(() => {
+    setChosenPage(currentPage);
+  }, [perPage]);
+
+  useEffect(() => {
+    onPageChange(chosenPage);
+  }, [chosenPage]);
```

Comment:
Why would you need that considering you have that state in the parent component?